File: frontend/src/types/application.ts
import { DatabaseConnectionSummary } from './database';

// Application Types
export interface Application {
  Id: string;
  Name: string;
  Description?: string;
  Version?: string;
  Environment?: string;
  Tags?: string;
  IsActive: boolean;
  CreatedAt: string;
  UpdatedAt: string;
  LastAccessedAt?: string;
  DatabaseConnectionCount: number;
  DatabaseConnections?: DatabaseConnectionSummary[];
}

export interface ApplicationRequest {
  Name: string;
  Description?: string;
  Version?: string;
  Environment?: string;
  Tags?: string;
  IsActive: boolean;
}

export interface ApplicationWithConnectionRequest {
  // Application details (matching backend's PascalCase expectations)
  ApplicationName: string;
  ApplicationDescription?: string;
  Version?: string;
  Environment?: string;
  Tags?: string;
  IsApplicationActive: boolean;
  
  // Connection details
  ConnectionName: string;
  ConnectionDescription?: string;
  DatabaseType: number;
  Server: string;
  Port?: number;
  Database?: string;
  Username?: string;
  Password?: string;
  ConnectionString?: string;
  ApiBaseUrl?: string;
  ApiKey?: string;
  AdditionalSettings?: string;
  IsConnectionActive: boolean;
  TestConnectionOnCreate?: boolean;
  
  // Cloud-specific fields
  AuthenticationMethod?: number;
  Region?: string;
  AccountId?: string;
  ProjectId?: string;
  InstanceId?: string;
  AccessKeyId?: string;
  SecretAccessKey?: string;
  SessionToken?: string;
  ClientId?: string;
  ClientSecret?: string;
  TenantId?: string;
  SubscriptionId?: string;
  CertificatePath?: string;
  CertificatePassword?: string;
  Scope?: string;
  Audience?: string;
  GrantType?: string;
  TokenEndpoint?: string;
  
  // Additional cloud fields
  ResourceGroup?: string;
  ServiceAccountKey?: string;
  RedirectUri?: string;
  PrivateKey?: string;
  PublicKey?: string;
  Account?: string;
  Warehouse?: string;
  Schema?: string;
  WorkspaceUrl?: string;
  ClusterId?: string;
  
  // GitHub-specific fields
  GitHubToken?: string;
  GitHubOrganization?: string;
  GitHubRepository?: string;
}

export interface ApplicationWithConnectionResponse {
  Application: Application;
  DatabaseConnection: DatabaseConnectionSummary;
  ConnectionTestResult?: boolean;
  ConnectionTestMessage?: string;
}

export interface ConnectionTestSchedule {
  Id?: string;
  ApplicationId: string;
  ApplicationName?: string;
  CronExpression: string;
  IsEnabled: boolean;
  LastRunTime?: string;
  NextRunTime?: string;
  LastRunStatus?: 'success' | 'failed' | 'running';
  CreatedAt?: string;
  UpdatedAt?: string;
}

export interface ConnectionTestScheduleRequest {
  ApplicationId: string;
  CronExpression: string;
  IsEnabled: boolean;
}

File: Backend/Model/Application.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace cams.Backend.Model
{
    public class Application
    {
        public Guid Id { get; set; } = Guid.NewGuid();

        [Required]
        public Guid UserId { get; set; }

        [Required]
        [StringLength(100)]
        public string Name { get; set; } = string.Empty;

        [StringLength(1000)]
        public string? Description { get; set; }

        [StringLength(50)]
        public string? Version { get; set; }

        [StringLength(200)]
        public string? Environment { get; set; }

        [StringLength(500)]
        public string? Tags { get; set; }

        public bool IsActive { get; set; } = true;

        public DateTime CreatedAt { get; set; }

        public DateTime UpdatedAt { get; set; }

        public DateTime? LastAccessedAt { get; set; }

        [ForeignKey("UserId")]
        public virtual User User { get; set; } = null!;

        public virtual ICollection<DatabaseConnection> DatabaseConnections { get; set; } = new List<DatabaseConnection>();
    }
}